governance/tsc/tsc-2024-04-29.md

```diff
@@ -0,0 +1,238 @@
+
+oriol: I'm thinking about blink there were various cases where if you keep nested, it became exponential. I think for LayoutNG they avoided this they used caching to avoid this. Maybe we can check what they are doing and try to adapt it into Servo.
+
+martin: that's not always the case, sometimes you have to do relayouts, depending on the layout mode (it happens sometime for floats and min|max-size situations)
+
+nico: So you might have to cache. The cache we had in taffy was 500 bytes per node. Maybe you can optimize that.
+
+nico: I might look into querying against the LayoutContext. I'll start by laying out and then seeing how it works? Does that seem okay?
+
+### Future of rust-harfbuzz crate
+
+martin: last year servo has over like 100 github repositories, that's a lot to maintain for a small team, we've going through and checking which are still active or use and archive the ones that aren't, combine then when it makes sense, a lot of times they were used only by one crate
+
+martin: there's a couple of interesting edge cases, rust-harfbuzz is one of them
+
+martin: it's a low level API, which servo uses; and a high level one, which servo doesn't use
+
+martin: and there's an official harfbuzz that provides the high level API
+
+martin: Behdad has say they'll be ok adopting things we cannot maintain anymore, would it make sense to give up ownership of these crates
+
+martin: one option is give up on the high level API we're not using, or give up in both of those and then the harfbuzz project would be responsibly also for maintaining it
+
+martin: another option could be transfer both, the issue might be if Servo needs a new version at some point
+
+manish: I'm happy to at least maintenance mode maintain the crate. I think that could work.
+
+manish: If we want harfbuzz-sys to be in Servo and to have one of them maintained by a broader team.
+
+martin: that raises another question, harfbuzz has its own high level bindings, not sure if they have a different philosophy
+
+nico: the harfbuzz one hasn't been updated for 9 months
+
+martin: our looks better now but there was a big gap anyway
+
+martin: ???
+
+manish: I like harfbuzz-rs, the crate we have, it's useful; if there's not a lot of effort to maintain it; there's not a big deal having two crates; the one we have works pretty well
+
+martin: are there features in the C API that are not yet in our high level bindings?
+
+nico: I wonder the fact they haven't updated their bindings, it means the API hasn't changed a lot recently
+
+manish: harfbuzz rarely changes the API
+
+martin: would it make sense to split?
+
+manish: I'd have it in the servo org, the harfbuzz org or in its own org
+
+manish: having an user in tree is useful
+
+martin: other projects (like freetype-sys) don't have it
+
+manish: that's a different case, and it's only used in Servo
+
+martin: I was referring to the one by ??? developers
```

Comment:
@mrobinson maybe you can fill the gap here? :innocent:

Reply:
```suggestion
martin: I was referring to the one by PistonDevelopers
```